src/components/pipeline/PipelineForm.js

```diff
@@ -0,0 +1,105 @@
+import React, { useEffect, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import PropTypes from 'prop-types';
+import { List } from 'immutable';
+
+import Grid from '@material-ui/core/Grid';
+import SaveIcon from '@material-ui/icons/Save';
+import Button from '@material-ui/core/Button';
+import TextField from '@material-ui/core/TextField';
+
+import {
+  EDIT_PIPELINE_SAVE_BUTTON_ID,
+  EDIT_PIPELINE_NAME_ID,
+  EDIT_PIPELINE_DESCRIPTION_ID,
+} from '../../config/selectors';
+
+import PipelineAccordion from './PipelineAccordion';
+
+const PipelineForm = ({
+  pipelineAlgorithms,
+  algorithms,
+  name,
+  description,
+  handleSave,
+}) => {
+  const { t } = useTranslation();
+
+  const [pipelineName, setPipelineName] = useState(name);
+  const [pipelineDescription, setPipelineDescription] = useState(description);
+
+  useEffect(() => {
+    setPipelineName(name);
+  }, [name]);
+
+  useEffect(() => {
+    setPipelineDescription(description);
+  }, [description]);
```

Comment:
Let's discuss during our call, but I believe these are redundant - you are already updating the state via `handleNameChange` and `handleDescriptionChange`, right?